crates/tabby-scheduler/Cargo.toml

```diff
@@ -14,7 +14,7 @@ tree-sitter-tags = "0.20.2"
 lazy_static = { workspace = true }
 tree-sitter-python = "0.20.2"
 tree-sitter-java = "0.20.2"
-tree-sitter-kotlin = "0.3.1"
+tree-sitter-kotlin = "=0.3.1"
```

Comment:
is this `=` necessary? what it represents?

Reply:
Yes, it's necessary. It pins the version of `tree-sitter-kotlin` to a specific version, because their upstream pushed a breaking change (I believe related to https://github.com/fwcd/tree-sitter-kotlin/commit/d5dc99a9bdc3fe895e6bcd39caddcfa4820f4c03) which causes it to use a different version of `tree-sitter` than us and the rest of the crates. However, this was done in a minor patch `0.3.6` instead of a major patch like `0.4.*`, meaning cargo treats it as a non-breaking change. When running `cargo update` (required to resolve another dependency conflict, with the `either` crate in two of our dependencies), it would select a version of `tree-sitter-kotlin` with this breaking change if the version is not pinned.

Reply:
Please add some comments on top of the line to explain the situation - we can revise it in future.